phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -674,6 +678,7 @@ private void generateMutations(final TableRef tableRef, final long mutationTimes
                         mutation.setAttribute(PhoenixIndexBuilder.ATOMIC_OP_ATTRIB, onDupKeyBytes);
                     }
                 }
+                annotateMutationsWithMetadata(table, rowMutations);
```

Comment:
perhaps move this to line 684? (out of if else?)

Reply:
Done.